include/ranking_response.h

```diff
@@ -59,124 +46,119 @@ namespace reinforcement_learning {
-     * @param model_id
-     */
-    void set_model_id(const char* model_id);
-
-    /**
-     * @brief Set the model_id.
-     * Every call to choose action is associated with a unique model used to predict.  A unique model_id
-     * is associated with each unique model. (This is set internally by the API).
-     * Input model_id is left in an unspecified but valid state.
-     * @param model_id
-     */
-    void set_model_id(std::string&& model_id);
+	/**
+	* @brief Get the chosen action id.
+	*
+	* @param action_id Chosen action id
+	* @param status Optional field with detailed string description if there is an error
+	* @return int Error code
+	*/
+	int get_chosen_action_id(size_t& action_id, api_status* status = nullptr) const; // id of the top action chosen by the model
+
+	/**
+  * @brief Set the chosen action id.  (This is set internally by the API)
+  *
+  * @param action_id Chosen action id
+  * @param status Optional field with detailed string description if there is an error
+  * @return int Error code
+  */
+	int set_chosen_action_id(size_t action_id, api_status* status = nullptr); // id of the top action chosen by the model
+
+	/**
+	* @brief Set the chosen action id, but do not verify the index fits within the ranking.  (This is set internally by the API)
+	* This is used in CCB where subsequent ranking_responses have subsets of the orignal actionset.
+	*
+	* @param action_id Chosen action id
+	* @param status Optional field with detailed string description if there is an error
+	* @return int Error code
+	*/
+	int set_chosen_action_id_unchecked(size_t action_id, api_status* status = nullptr);
+
+	/**
+	* @brief Add (action id, probability) pair to the slot (This is set internally by the API)
+	*
+	* @param action_id
+	* @param prob
+	*/
+	void push_back(const size_t action_id, const float prob);
+
+	/**
+	* @brief Size of the action collection.
+	*
+	* @return size_t
+	*/
+	size_t size() const;
+
+	/**
+	* @brief Set the model_id.
+	* Every rank call is associated with a unique model used to predict.  A unique model_id
+	* is associated with each unique model. (This is set internally by the API)
+	* @param model_id
+	*/
+	void set_model_id(const char* model_id);
+
+	/**
+	* @brief Set the model_id.
+	* Every rank call is associated with a unique model used to predict.  A unique model_id
+	* is associated with each unique model. (This is set internally by the API).
+	* Input model_id is left in an unspecified but valid state.
+	* @param model_id
+	*/
+	void set_model_id(std::string&& model_id);
+
+	/**
+	* @brief Get the model_id.
+	* Every rank call (single or multi slot) is associated with a unique model used to predict.  A unique model_id
+	* is associated with each unique model. (This is set internally by the API)
+	* @return const char*
+	*/
+	const char * get_model_id() const;
+
+	/**
+	* @brief Clear the slot detailed object so that it can be reused.
```

Comment:
looks like all comments here should stay the same - slot_ranking is implementation detail from this class point of view